purchase_ext_sst/models/purchase_order.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2017-2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    address = fields.Char()
+    remark = fields.Text()
+    date_planned = fields.Datetime(compute=False)
+    sale_prediction_amount = fields.Monetary("Sales Prediction")
+
+    def button_confirm(self):
+        for order in self:
+            if self.is_default_partner(order.partner_id.id):
+                raise UserError(
+                    _("Purchase order cannot be confirmed with default guest user.")
+                )
+            if not order.date_planned:
+                order.date_planned = fields.Datetime.now()
+                for order_line in order.order_line:
+                    order_line.date_planned = order.date_planned
+        return super(PurchaseOrder, self).button_confirm()
+
+    def is_default_partner(self, partner_id):
+        company = self.env.user.company_id
+        default_id = (
+            self.env["ir.default"].get(
+                "purchase.order",
+                "partner_id",
+                user_id=self.env.uid,
+                company_id=company.id,
+            )
+            or self.env["ir.default"].get(
+                "purchase.order", "partner_id", user_id=False, company_id=company.id
+            )
+            or False
+        )
+        return partner_id == default_id
```

Comment:
```suggestion
    def is_default_partner(self, partner_id):
        company = self.env.user.company_id
        return partner_id == company.purchase_default_partner_id
```
I didn't think it was necessary to use `ir.default` in the logic to identify the default partner.
What about an idea that has a record of the partner to be set up for the company and makes it the default? 
Could you please check the status of other modules that depend on `is_default_partner` ?